warp10/src/main/java/io/warp10/script/functions/FETCH.java

```diff
@@ -1099,4 +984,197 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
 
     return params;
   }
+
+  /**
+   * Compute the time range given start, end, timespan and count. One of these 3 parameters must be null.
+   * @param start A Long, a String representing a Long, a String representing an ISO8601 date or "now". This represents the start of the time range.
+   * @param startParamName The name of the start parameter, for error generation.
+   * @param end A Long, a String representing a Long, a String representing an ISO8601 date or "now". This represents the end of the time range.
+   * @param endParamName The name of the end parameter, for error generation.
+   * @param timespan A Long, a String representing a Long or a String representing an ISO8601 duration. This represents the duration of the time range.
+   * @param timespanParamName The name of the timespan parameter, for error generation.
+   * @param count A positive Long. This represents the number of point to return in the time range.
+   * @param countParamName The name of the count parameter, for error generation.
+   * @return An array of three Longs: [start, end, timespan] with the guarantee that start and end are not null.
+   * @throws WarpScriptException If the time range specification is invalid.
+   */
+  public static Long[] computeTimeRange(Object start, String startParamName, Object end, String endParamName, Object timespan, String timespanParamName, Object count, String countParamName) throws WarpScriptException {
+    long now = TimeSource.getTime();
+
+    //
+    // Try to convert start to a valid timestamp if possible.
+    //
+
+    Long startTs = null;
+
+    if (start instanceof Long) {
+      // Simple case: start is a long
+      startTs = (long) start;
+    } else if (start instanceof String) {
+      // If it's a string, it may be the string representation of a Long, a ISO8601 date or 'now'.
+      if ("now".equals(start)) {
+        startTs = now;
+      } else {
+        try {
+          startTs = Long.valueOf((String) start);
```

Comment:
Could use Long.parseLong

Reply:
In this case this is totally equivalent. But in case `startTs` is changed to a `long` in the future, `Long.parseLong` should be preferred.